src/contexts/LanguageContext.tsx

```diff
@@ -0,0 +1,82 @@
+"use client";
+
+import React, { createContext, useContext, useState, useEffect, ReactNode } from "react";
+import {
+  TargetLanguageCode,
+  LanguageOption,
+  SUPPORTED_LANGUAGES,
+  DEFAULT_TARGET_LANGUAGE,
+  LANGUAGE_STORAGE_KEY,
+  VALID_LANGUAGE_CODES,
+} from "@/utils/languages";
+
+type LanguageContextType = {
+  targetLanguage: TargetLanguageCode;
+  setTargetLanguage: (code: TargetLanguageCode) => void;
+  availableLanguages: readonly LanguageOption[];
+};
+
+const LanguageContext = createContext<LanguageContextType | undefined>(undefined);
+
+type LanguageProviderProps = {
+  children: ReactNode;
+};
+
+export function LanguageProvider({ children }: LanguageProviderProps) {
+  const [targetLanguage, setTargetLanguageState] = useState<TargetLanguageCode>(
```

Comment:
This doesn't follow naming conventions. It should be `const [targetLanguage, setTargetLanguage]`